### PR TITLE
Add capabilities: 1) bind tidb-dashboard only to localhost; 2) disable keyVisualizer

### PR DESCRIPTION
**Pull request by smineyev81**
### What problem does this PR solve?
Adds abilities to make tidb-dashboard only accessible via port-forwarding and disable keyVisualizer

### What is changed and how does it work?
1) new tidb-dashboard CRD fields: ListenOnLocalhostOnly and EnableKeyVisualizer; 
2) bind tidb-dashboard to localhost only if ListenOnLocalhostOnly is true and disables key visualizer if EnableKeyVisualizer=false

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [x] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```
NONE
```

### CODE REVIEW

File: cmd/http-service/go.sum [hunk 1]
```diff
@@ -1,34 +1,177 @@
+cloud.google.com/go v0.110.6/go.mod h1:+EYjdK8e5RME/VY/qLCAtuyALQ9q67dvuum8i+H5xsI=
```
Comment (csuzhangxc) [inline cmd/http-service/go.sum:1]: after run `go mod tidy` in the master branch, I didn't see these diff.

**Review by csuzhangxc — COMMENTED** [review]
what's the scenario that we need to bind to localhost only

**Comment by smineyev81** [comment]
> Adds ability to make tidb-dashboard only accessible via port-forwarding

For security reason we (Airbnb) do not want to expose tidb-dashboard outside. It should be only accessible by infra engineers via port-forwarding

**Comment by smineyev81** [comment]
@csuzhangxc , I added one more feature flag into this PR.
Please re-review

**Comment by csuzhangxc** [comment]
> > Adds ability to make tidb-dashboard only accessible via port-forwarding
> 
> For security reason we (Airbnb) do not want to expose tidb-dashboard outside. It should be only accessible by infra engineers via port-forwarding

Got it.

Please fix ci / pull (test)

**Comment by smineyev81** [comment]
> @csuzhangxc

@csuzhangxc , fixed. Please re-review

File: pkg/apis/pingcap/v1alpha1/tidbdashboard_types.go [hunk 2]
```diff
@@ -107,6 +107,17 @@ type TidbDashboardSpec struct {
...
+	//ListenOnLocalhostOnly whether to expose dashboard to 0.0.0.0 or limit it to localhost only
+	// which means it will be accessible only via port-forwarding
+	// Optional: Defaults to false
+	// +optional
+	ListenOnLocalhostOnly *bool `json:"listenOnLocalhostOnly,omitempty" default:"false"`
+
+	// KeyVisualizer is whether to enable Key Visualizer.
+	// Optional: Defaults to true
+	// +optional
+	KeyVisualizer *bool `json:"keyVisualizer,omitempty" default:"true"`
```
Comment (csuzhangxc) [inline pkg/apis/pingcap/v1alpha1/tidbdashboard_types.go:120]: How about changing it to `DisableKeyVisualizer` with the `false` default?

Comment (smineyev81) [reply pkg/apis/pingcap/v1alpha1/tidbdashboard_types.go]: @csuzhangxc, done

File: pkg/manager/tidbdashboard/manager.go [hunk 3]
```diff
@@ -414,22 +428,24 @@ func generateTiDBDashboardService(td *v1alpha1.TidbDashboard) *corev1.Service {
...
-		"-h=0.0.0.0",
+		fmt.Sprintf("-h=%s", listenHost),
 		fmt.Sprintf("-p=%d", port),
 		fmt.Sprintf("--data-dir=%s", dataPVCMountPath),
 		fmt.Sprintf("--temp-dir=%s", dataPVCMountPath),
 		fmt.Sprintf("--path-prefix=%s", pathPrefix),
 		fmt.Sprintf("--feature-version=%s", featureVersion),
 		fmt.Sprintf("--experimental=%t", experimental),
 		fmt.Sprintf("--telemetry=%t", telemetry),
+		fmt.Sprintf("--keyVisualizer=%t", keyVisualizer),
```
Comment (csuzhangxc) [inline pkg/manager/tidbdashboard/manager.go:448]: how about only appending this arg if it needs to be disabled, then we can avoid recreating the existing Pod when upgrading the TiDB Operator.

Comment (smineyev81) [reply pkg/manager/tidbdashboard/manager.go]: @csuzhangxc, done. Please re-review

**Review by csuzhangxc — APPROVED** [review]

**Pull request merged**